cmd/mailchain/internal/settings/mailbox_state.go

```diff
@@ -1,44 +1,60 @@
 package settings
 
 import (
+	"fmt"
+
 	"github.com/mailchain/mailchain/cmd/mailchain/internal/settings/defaults"
 	"github.com/mailchain/mailchain/cmd/mailchain/internal/settings/output"
 	"github.com/mailchain/mailchain/cmd/mailchain/internal/settings/values"
 	"github.com/mailchain/mailchain/stores"
+	"github.com/mailchain/mailchain/stores/bdbstore"
 	"github.com/mailchain/mailchain/stores/ldbstore"
-	"github.com/pkg/errors"
 )
 
+// State interface wraps basic methods required for mailbox state.
+type State interface {
+	Produce() (stores.State, error)
+	Output() output.Element
+	Kind() values.String
+}
+
+// mailboxState returns the default MailboxState which uses leveldb store.
 func mailboxState(s values.Store) *MailboxState {
-	k := &MailboxState{
-		Kind:                values.NewDefaultString(defaults.MailboxStateKind, s, "mailboxState.kind"),
-		mailboxStateLevelDB: mailboxStateLevelDB(s),
+	return &MailboxState{
+		mailboxState: mailboxStateLevelDB(s),
+	}
+}
```

Comment:
This should read something like this

```
func mailboxState(s values.Store) *MailboxState {
	k := &MailboxState{
		Kind:                values.NewDefaultString(defaults.MailboxStateKind, s, "mailboxState.kind"),
		mailboxStateLevelDB: mailboxStateLevelDB(s),
                 mailboxStateBadgerDB: mailboxStateBadgerDB(s),
	}
	return k
}
```

This will init with all the settings for each time then `kind` will choose which to return.